rewatch/src/build/packages.rs

```diff
@@ -515,17 +515,22 @@ pub fn get_source_files(
     };
 
     let path_dir = Path::new(&source.dir);
-    match (build_dev_deps, type_) {
-        (false, Some(type_)) if type_ == "dev" => (),
+    let is_clean = match dev_deps {
+        DevDeps::Clean => true,
+        _ => false,
+    };
+    match (dev_deps, type_) {
+        (DevDeps::DontBuild, Some(type_)) if type_ == "dev" => (),
         _ => match read_folders(filter, package_dir, path_dir, recurse) {
             Ok(files) => map.extend(files),
 
-            Err(_e) => log::error!(
+            Err(_e) if !is_clean => log::error!(
```

Comment:
why would this ever happen, `dev_deps` is by definition `DontBuild` here, and `is_clean` is defined as `dev_deps` being `DevDeps::Clean`

Reply:
Right now, clean is trying to remove files that are listed as `type: "dev"`.
This happens for third-party packages inside the node_modules.
These files can be absent when published to npm.
Test files are a good example of that: https://github.com/rescript-lang/experimental-rescript-webapi/blob/d1645e31cd4f46e1b3a1bed0156e7b6930f8ea5d/rescript.json#L10-L14

Reply:
I know, butI don't think your answer answers my question. When is this arm actually hit, I think there are no cases.